.github/workflows/ci.yml

```diff
@@ -0,0 +1,78 @@
+name: Build & Tests
+
+on:
+  push:
+  pull_request:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build_test:
+    # Used for installing Miri
+    # https://rust-lang.github.io/rustup-components-history/
+    env:
+      TOOLCHAIN_VERSION: '2022-09-26'
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        channel: [ "1.56.1", "stable", "beta", "nightly" ]
+        target: [ "i686-unknown-linux-gnu", "x86_64-unknown-linux-gnu", "arm-unknown-linux-gnueabi", "aarch64-unknown-linux-gnu", "powerpc-unknown-linux-gnu", "powerpc64-unknown-linux-gnu", "wasm32-wasi" ]
+        features: [ "" , "--features alloc,simd,simd-nightly"]
+
+    # Prevents a workflow run from failing when a job fails
+    # If target is x86 or x86_64, 
+    # or we are using simd-nightly while not on the nightly channel
+    # do not continue on error (so crash on error)
+    continue-on-error: ${{ !(contains(matrix.target, 'x86_64') || contains(matrix.target, 'i686')) }}
+    # Use this instead when https://github.com/rust-lang/rust/issues/48556#issuecomment-1264564803 is resolved
+    # continue-on-error: ${{ !(contains(matrix.target, 'x86_64') || contains(matrix.target, 'i686')) || ((contains(matrix.features, 'simd-nightly')) && (matrix.channel != 'nightly')) }}
+
+    name: Build & Test (${{ matrix.channel }} for ${{ matrix.target }}, features set to ${{ matrix.features != '' }})
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Setup Rust ${{ matrix.channel }} ${{ matrix.target }}
+      uses: ATiltedTree/setup-rust@v1
+      with:
+        rust-version: ${{ matrix.channel }}
+        targets: ${{ matrix.target }}
+    
+    - name: Rust Cache
+      uses: Swatinem/rust-cache@v2.0.0
+      with:
+        key: ${{ matrix.channel }}-${{ matrix.target }}-${{ matrix.features == '' }}-${{ hashFiles('**/Cargo.lock') }}
+
+    - name: Check ${{ matrix.channel }}
+      run: cargo +${{ matrix.channel }} check --target ${{ matrix.target }} ${{ matrix.features }} --verbose
+
+    - name: Build ${{ matrix.channel }}
+      run: cargo +${{ matrix.channel }} build --target ${{ matrix.target }} ${{ matrix.features }} --verbose
+
+    # Only run tests in the x86 architecture
+    - name: Run ${{ matrix.channel }} tests
+      run: cargo +${{ matrix.channel }} test --target ${{ matrix.target }} ${{ matrix.features }} --verbose
+      if: ${{ contains(matrix.target, 'x86_64') || contains(matrix.target, 'i686') }}
+
+    - name: Check README consistency
```

Comment:
Doesn't this currently fail? Should we comment it out until it succeeds?

Reply:
You mean the test or the README stuff?

Reply:
Yeah